plugins/ralph-specum/hooks/scripts/stop-watcher.sh

```
@@ -360,3 +763,249 @@ find "$CWD/$SPEC_PATH" -name ".progress-task-*.md" -mmin +60 -delete 2>/dev/null
+## Recovery
+1. Check filesystem: mount | grep "$(df "$spec_dir" | tail -1 | awk \"{print \\$1}\")"
+2. If read-only: remount rw: sudo mount -o remount,rw /path/to/mount
+3. Or resume with /ralph-specum:cancel and re-run in a writable environment'
+    printf '{"decision":"block","reason":"Filesystem heartbeat failed 2nd time — filesystem may be degraded","filesystemHealthFailures":%d,"spec":"%s","recoveryInstructions":"%s"}\n' "$failures" "$SPEC_NAME" "$(echo "$recovery_instructions" | tr '\n' ' ' | sed 's/  */ /g')"
+    exit 0
+  else
+    # 3rd+ failure: full block and exit
+    printf '{"decision":"block","reason":"Filesystem heartbeat failed %d consecutive times — full filesystem block","filesystemHealthFailures":%d,"spec":"%s"}\n' "$failures" "$failures" "$SPEC_NAME"
+    exit 0
+  fi
+}
```

Shellcheck (0.11.0)

[info] Command appears to be unreachable. Check usage (or ignore if invoked indirectly).

(SC2317)